### PR TITLE
feat: SwissTransfer Attachments

### PR DESCRIPTION
**Pull request by Lnamw**
- Add SwissTransfer attachments to the list of attachments (cells)
- Add text to specify number of attachments and number of files send via SwissTransfer 
- On clic on SwissTransfer attachments open DocumentPicker to save in File
- Update downloadAll method to save 2 zip files: one for attachment and a second one for SwissTransfer attachment and open DocumentPicker to save in File

Capture d’écran 2024-06-10 à 17 18 03

### CODE REVIEW

File: Mail/Views/Thread/Message/Attachment/AttachmentsView.swift [hunk 1]
```diff
@@ -43,51 +43,74 @@ struct AttachmentsView: View {
...
-                            AttachmentCell(attachment: attachment)
+                            AttachmentView(title: attachment.name, subtitle: attachment.size.formatted(.defaultByteCount), icon: attachment.icon)
+                        }
+                    }
+                    if let swissTransferAttachment = message.swissTransferAttachment {
+                        ForEach(swissTransferAttachment.files) { file in
+                            Button {
+                                downloadSwissTransferAttachment(stUuid: swissTransferAttachment.uuid, fileUuid: file.uuid)
+                            } label: {
+                                AttachmentView(title: file.name, subtitle: file.size.formatted(.defaultByteCount), icon: file.icon)
```
Comment (valentinperignon) [inline Mail/Views/Thread/Message/Attachment/AttachmentsView.swift:61]: Maybe we can have a AttachmentView(attachment: Attachment...) and a AttachmentView(swissTransferFile: ...) to simplify the code

File: Mail/Views/Thread/Message/Attachment/AttachmentsView.swift [hunk 2]
```diff
@@ -43,51 +43,74 @@ struct AttachmentsView: View {
...
         }
-        .sheet(item: $allAttachmentsURL) { allAttachmentsURL in
-            DocumentPicker(pickerType: .exportContent([allAttachmentsURL.url]))
+        .sheet(item: $attachmentsURL) { attachmentsURL in
+            DocumentPicker(pickerType: .exportContent(attachmentsURL.urls))
                 .ignoresSafeArea()
         }
     }
 
+    private func text() -> String {
```
Comment (valentinperignon) [inline Mail/Views/Thread/Message/Attachment/AttachmentsView.swift:103]: As discussed, we can try to format a list

File: Mail/Views/Thread/Message/Attachment/AttachmentsView.swift [hunk 3]
```diff
@@ -40,48 +40,74 @@ struct AttachmentsView: View {
...
+        }
+        if let swissTransferAttachment = message.swissTransferAttachment {
+            text.append("\(MailResourcesStrings.Localizable.fileQuantity(swissTransferAttachment.nbfiles))")
+        }
+        return text.formatted(.list(type: .and))
+    }
+
+    private var formattedSize: String {
+        guard let swissTransferAttachment = message.swissTransferAttachment
+        else { return message.attachmentsSize.formatted(.defaultByteCount) }
```
Comment (valentinperignon) [inline Mail/Views/Thread/Message/Attachment/AttachmentsView.swift:56]:
```suggestion
        guard let swissTransferAttachment = message.swissTransferAttachment else {
                return message.attachmentsSize.formatted(.defaultByteCount)
        }
```

File: MailCore/Models/SwissTransferAttachment.swift [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+ */
+
+import Foundation
+import MailResources
+import RealmSwift
+
+public class SwissTransferAttachment: EmbeddedObject, Codable {
+    @Persisted public var uuid: String
+    @Persisted public var nbfiles: Int
+    @Persisted public var size: Int64
```
Comment (valentinperignon) [inline MailCore/Models/SwissTransferAttachment.swift:26]: Why should we precise Int64? As Swift Int are already 64 bits

Comment (Lnamw) [reply MailCore/Models/SwissTransferAttachment.swift:26]: No particular reason, just to be consistent with the Attachment model

**Pull request merged**